### ISSUE TITLE
Datasource output breaks if textarea is included

### ISSUE THREAD

**Issue by michael-e** [symphonists/subsectionmanager#1]
If you include a textarea (which uses a text formatter) in the datasource output, page rendering will break with the following error:

```
Missing argument 4 for fieldTextarea::appendFormattedElement(), called in /Users/me/Sites/dev/me/symphony207/extensions/subsectionmanager/fields/field.subsectionmanager.php on line 509 and defined
/Users/me/Sites/dev/me/symphony207/symphony/lib/toolkit/fields/field.textarea.php line 166

161     
162     private function __replaceAmpersands($value) {
163         return preg_replace('/&(?!(#[0-9]+|#x[0-9a-f]+|amp|lt|gt);)/i', '&', trim($value));
164     }
165     
166     public function appendFormattedElement(&$wrapper, $data, $encode = false, $mode) {
167         
168         if ($mode == null || $mode == 'formatted') {
169             
170             if ($this->get('formatter') && isset($data['value_formatted'])) {
```

The problem seems to be that in line 509 of `field.subsectionmanager.php` the `$mode` argument is missing. So if you replace this line for example by 

```
$field->appendFormattedElement($item, $values, false, NULL);
```

it works - but I am **completely unsure** about the correct $mode value!!!

**Comment by nilshoerrmann** [comment]
I think the `$mode` is used for output simulation. But I'm not sure. Has something changed in this area of Symphony?!

**Comment by michael-e** [comment]
I think is has to do with formatted vs. unformatted output. For backward compatibility Alistair even included this in line 168 of `field.textarea.php`:

```
if ($mode == null || $mode == 'formatted') {
    ...
}
```

This means that you have decide if you want to output the formatted version or not. It would probably be the best to look up the desired output in the related section and set the parameter accordingly (formatted/unformatted, I guess).

**Comment by nilshoerrmann** [comment]
Ah yes, that makes sense. Subsection Manager isn't backwards compatible at all so we should not have to care about that. If this is about the formatted and unformatted setting we should start thinking about a way how to integrate the included elements settings directly into the data source editor.

**Comment by michael-e** [comment]
What about displaying them together with their section name? Let's say we have a section called "media":
- media: name
- media: file
- media: description

**Comment by michael-e** [comment]
Sorry, it must be like this:
- media: name
- media: file
- media: description: formatted
- media: description: unformatted

**Comment by nilshoerrmann** [comment]
That's a good idea - or their field name. E. g. a Subsection Manager field with the name 'photos' (linking to your media section):
- photos: name
- photos: file
- photos: description: unformatted
- photos: description: formatted

**Comment by michael-e** [comment]
Exactly.
- section name: field name: formatting (if applicable)

**Comment by nilshoerrmann** [comment]
I think I'll have to investigate how the textarea field handles the implementation of these different types.

**Comment by nickdunn** [comment]
The Textarea is broken in that its appendFormattedElement doesn't have a default set for `$mode`. All other fields do. I believe this has been fixed in 2.0.7 or 2.0.8, so it's not actually a bug with Subsection.

**Comment by michael-e** [comment]
Looks like it was fixed in 2.0.8. So extension compatibility might be changed to 2.0.8+.

**Comment by nilshoerrmann** [comment]
Great - thanks for the feedback.
